dagql/server.go

```diff
@@ -480,33 +489,9 @@ func (s *Server) Select(ctx context.Context, self Object, dest any, sels ...Sele
 			} else if isEnum {
 				// HACK: list of objects must be the last selection right now unless nth used in Selector.
 				if i+1 < len(sels) {
-					return fmt.Errorf("cannot sub-select enum of %s", res.Type())
-				}
-				for nth := 1; nth <= enum.Len(); nth++ {
-					val, err := enum.Nth(nth)
-					if err != nil {
-						return fmt.Errorf("nth %d: %w", nth, err)
-					}
-					if wrapped, ok := val.(Derefable); ok {
-						val, ok = wrapped.Deref()
-						if !ok {
-							if err := appendAssign(reflect.ValueOf(dest).Elem(), nil); err != nil {
-								return err
-							}
-							continue
-						}
-					}
-					nthID := id.Clone()
-					nthID.SelectNth(nth)
-					obj, err := s.toSelectable(nthID, val)
-					if err != nil {
-						return fmt.Errorf("select %dth array element: %w", nth, err)
-					}
-					if err := appendAssign(reflect.ValueOf(dest).Elem(), obj); err != nil {
-						return err
-					}
+					return fmt.Errorf("cannot sub-select enumerable: %s", res.Type())
 				}
-				return nil
+				return assign(reflect.ValueOf(dest).Elem(), res)
```

Comment:
@sipsma I mentioned this could probably be simplified before, decided to do that now but lmk if there was a reason you couldn't (seems OK judging by tests)

Reply:
I think I left it as is because it wasn't 100% clear to me that it was doing all the exact same things. E.g. the previous code appended the `obj` from `s.ToSelectable(nthID, val)` but afaict just using `assign` won't end up doing that (based on the implementation of `DynamicArrayOutput.SetField`).

Really not sure whether that's meaningful and, if so, in which cases, so no biggie either way at the moment. Can patch up more if we hit a case where it breaks.

Reply:
Ah, it looks like nothing actually calls this path right now because the whole conditional is beneath `if _, ok := s.ObjectType(res.Type().Name()); ok {` and the only slice value we ever select out is `[]string`, which is not an object type. I'll just revert this to reduce noise.

Reply:
We do also select into `[]dagql.Instance[*core.ModuleDependency]` here: https://github.com/sipsma/dagger/blob/58f4b8f7791786a740134ee36854002f9021c1fc/core/schema/module.go#L686-L686

But sounds good nonetheless